### ISSUE TITLE
File reading works differently depending on OS

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1580]
**Reported by sjoelund&#46;se on 6 Oct 2014 14:04 UTC**
We have in the code for ModelicaInternal.c:
```c
fp = fopen(fileName, "r");
```
Which works differently on Windows and Linux (copyFile works the same on both OS since "rb" is used there). This is a problem because the `readLine` routines will produce a different result on Linux:
```mo
{"line1\r","line2\r",...} // Linux
{"line1","line2",...} // Windows
```

There are two ways to solve this: make file reading strip "\r" at the end of a line silently. Or keep it on all OS'es. I would suggest stripping it since we have found some third-party code that depends on \r being translated.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1580

**Comment by modelica-trac-importer** [comment]
**Modified by sjoelund&#x2e;se on 6 Oct 2014 14:04 UTC**

**Comment by modelica-trac-importer** [comment]
**Modified by beutlich on 7 Oct 2014 08:01 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 7 Oct 2014 08:13 UTC**
attachment:T1580_maint.patch fixes the problem (for maintenance branch). Can you please confirm! Thanks.

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#46;se on 7 Oct 2014 08:26 UTC**
It seems to work correctly

```
marsj@marsj-laptop:~/tmp$ cat a.mos
loadModel(Modelica);getErrorString();
Modelica.Utilities.Streams.readFile("abc.txt");getErrorString();

marsj@marsj-laptop:~/tmp$ cat abc.txt
abc
def
ghi

marsj@marsj-laptop:~/tmp$ file abc.txt
abc.txt: ASCII text, with CRLF line terminators

marsj@marsj-laptop:~/tmp$ omc-with-patch a.mos
true
""
{"abc","def","ghi"}
""

marsj@marsj-laptop:~/tmp$ omc-nightly a.mos
true
""
"}"ghi
""
```

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#46;se on 7 Oct 2014 08:28 UTC**
I will note though: The allocated string is not the same length as the returned string. Would be good to fix that.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 7 Oct 2014 09:10 UTC**
Replying to [comment:5 sjoelund.se]:
> I will note though: The allocated string is not the same length as the returned string. Would be good to fix that.

OK, please try attachment:T1580_2_maint.patch.

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 7 Oct 2014 11:00 UTC**
Yes, this works and looks fine to me.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 7 Oct 2014 13:56 UTC**
Resolved by bbcfe6855320eceb435ee2a7255e5c9457a8f211/ d178caabd8d6770442600436f324ba14cc7172d3 (maint. branch) and 9b7fef32de5eba6185b59355c6ac46c65b7f6644 (trunk).

**Comment by modelica-trac-importer** [comment]
**Changelog modified by beutlich on 7 Oct 2014 13:56 UTC**
Fixed platform dependency of ModelicaInternal_readLine

**Comment by modelica-trac-importer** [comment]
**Comment by hansolsson on 13 Oct 2014 15:49 UTC**
The function ModelicaInternal_readFile now behaves differently from ModelicaInternal_readLine - that doesn't seem good.

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 13 Oct 2014 15:57 UTC**
I think that is fine. ModelicaInternal_readFile is unused in MSL 1.6, 2.2.2, 3.1, 3.2.1, and trunk. Maybe the code should even be removed.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 13 Oct 2014 20:06 UTC**
Replying to [comment:10 sjoelund.se]:
> I think that is fine. ModelicaInternal_readFile is unused in MSL 1.6, 2.2.2, 3.1, 3.2.1, and trunk. Maybe the code should even be removed.

But it is used in ModelicaTest.

**Comment by modelica-trac-importer** [comment]
**Comment by sjoelund&#x2e;se on 13 Oct 2014 20:09 UTC**
Is the external "C" function used? Or is Streams.readFile called? (Streams.readFile is implemented as Modelica code calling ModelicaInternal_readLine; it does not use ModelicaInternal_readFile)

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 13 Oct 2014 20:13 UTC**
You are right indeed. ModelicaInternal_readFile is not used at all. Sorry for confusion.

Anyway, should be easy to apply patch also to ModelicaInternal_readFile.

**Comment by modelica-trac-importer** [comment]
**Comment by beutlich on 14 Oct 2014 07:29 UTC**
Also resolved by 9489a0e932d88f8f6fbd105882a810eff6740b97/ 9a8d452d8fb13deccf858b06703bd1850a1056c2 (maint. branch) and e7ba903f2246748b062f102f5ae4dc2042a68edf (trunk) for readFile.

**Comment by modelica-trac-importer** [comment]
**Changelog modified by beutlich on 14 Oct 2014 07:29 UTC**
Fixed platform dependency of ModelicaInternal_readFile and ModelicaInternal_readLine
